apps/test/unit/lib/util/audioApiTest.js

```diff
@@ -100,5 +104,88 @@ describe('Audio API', function() {
         language: 'English'
       });
     });
+
+    describe('block functionality', function() {
+      let outputWarningSpy, azureTTSStub, appOptions, options;
+
+      beforeEach(function() {
+        outputWarningSpy = sinon.spy();
+        injectErrorHandler({outputWarning: outputWarningSpy});
+        azureTTSStub = {
+          createSoundPromise: sinon.spy(),
+          enqueueAndPlay: sinon.spy()
+        };
+        sinon.stub(AzureTextToSpeech, 'getSingleton').returns(azureTTSStub);
+        appOptions = {
+          azureSpeechServiceToken: 'fake-token',
+          azureSpeechServiceUrl: 'https://fake.tts.url',
+          azureSpeechServiceVoices: {
+            English: {female: 'en-female', languageCode: 'en-US'}
+          }
+        };
+        setAppOptions(appOptions);
+        options = {
+          text: 'hello world',
+          gender: 'female',
+          language: 'English'
+        };
+      });
+
+      afterEach(function() {
+        AzureTextToSpeech.getSingleton.restore();
+      });
+
+      it('no-ops if appOptions.azureSpeechServiceToken is missing', async function() {
+        delete appOptions.azureSpeechServiceToken;
+        setAppOptions(appOptions);
+        await commands.playSpeech(options);
+
+        expect(outputWarningSpy).to.have.been.calledOnce;
```

Comment:
tiny nit (feel free to ignore): this doesn't seem exactly like a no-op, since we are outputting a warning (ie. doing something). Consider renaming to "doesn't call azure" or something like that?

Reply:
yeah, it felt a little weird to keep calling this a "no-op" so i'll update my language here